OrbitCore/CrashHandler.cpp

```diff
@@ -10,89 +10,45 @@
 #include "TcpClient.h"
 
 #ifdef _WIN32
-#include <client/windows/handler/exception_handler.h>
-
-#include <fstream>
-
-//-----------------------------------------------------------------------------
-void SendDumpInternal(const std::wstring& a_Dir, const std::wstring& a_Id);
-google_breakpad::ExceptionHandler* GHandler;
+#include <Shlwapi.h>
 
 //-----------------------------------------------------------------------------
-bool DmpFilter(void* context, EXCEPTION_POINTERS* exinfo,
-               MDRawAssertionInfo* assertion) {
-  return true;
+CrashHandler::CrashHandler(const std::wstring& dump_path) {
+  bool not_init = false;
+  if (is_init_.compare_exchange_strong(not_init, true)) {
+    Init(dump_path);
+  }
 }
 
-//-----------------------------------------------------------------------------
-bool DmpCallback(const wchar_t* dump_path, const wchar_t* minidump_id,
-                 void* context, EXCEPTION_POINTERS* exinfo,
-                 MDRawAssertionInfo* assertion, bool succeeded) {
-  PRINT_FUNC;
+void CrashHandler::Init(const std::wstring& dump_path) {
+  // Creates a new CrashpadClient instance that directs crashes to crashpad
+  // handler. Minidump files will be written to dump_path.
 
-  std::string dir = Path::GetDumpPath();
-  std::string msg = "A crash dump was generated in " + dir;
-  MessageBox(NULL,
-             succeeded ? s2ws(msg).c_str() : L"Failed to generate crash dump",
-             L"Orbit Crash Handler: ", MB_ICONEXCLAMATION | MB_OK);
+  TCHAR exe_path[MAX_PATH];
+  GetModuleFileName(NULL, exe_path, MAX_PATH);
+  PathRemoveFileSpec(exe_path);
+  PathAppend(exe_path, L"crashpad_handler.exe");
```

Comment:
One suggestion here: What about making the path to crashpad_handler.exe a constructor argument?

If it is a constructor argument we have to provide the path in `OrbitQt/main.cpp`.
But that's easy because we can use just the Qt framework:
```
QDir{QCoreApplication::applicationDirPath()}.absoluteFilePath("crashpad_handler.exe").toStdString()
```
no win32-API and it should be platform-agnostic (almost, we still need an if for the `.exe`).

Reply:
Yes, I should do this along with moving dump_path to constructor argument.